bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/VisitDocumentController.java

```diff
@@ -61,16 +63,25 @@ public VisitDocumentResponse save(@RequestBody VisitDocumentRequest visitDocumen
     @RequestMapping(method = RequestMethod.POST, value = baseVisitDocumentUrl + "/uploadDocument")
     @ResponseBody
     public HashMap<String, String> saveDocument(@RequestBody Document document) {
-        Patient patient = Context.getPatientService().getPatientByUuid(document.getPatientUuid());
-        String encounterTypeName = document.getEncounterTypeName();
-        if (StringUtils.isEmpty(encounterTypeName)) {
-            encounterTypeName = administrationService.getGlobalProperty("bahmni.encounterType.default");
+        try {
+            HashMap<String, String> savedDocument = new HashMap<>();
+            Patient patient = Context.getPatientService().getPatientByUuid(document.getPatientUuid());
+            String encounterTypeName = document.getEncounterTypeName();
+            if (StringUtils.isEmpty(encounterTypeName)) {
+                encounterTypeName = administrationService.getGlobalProperty("bahmni.encounterType.default");
+            }
+            String fileName = document.getFileName();
+            fileName = fileName==null ? "" : fileName;
+            fileName = fileName.trim().replaceAll(" ", "-");
+            Paths.get(fileName);
+            fileName = fileName.replaceAll("__", "_");
```

Comment:
SPR violation... I would extract this out into a function

```
private String sanitizeFilename(String fileName) {
        if (filename == null) return ""
        else return filename.trim().replaceAll(" ", "-").replaceAll("__", "_");
    }
```

Reply:
Updated as per review comments. Please check.